### PR TITLE
use custmized version

### PR DESCRIPTION
**Pull request by XuanYang-cn**

### CODE REVIEW

**Comment by XuanYang-cn** [comment]
image

**Comment by XuanYang-cn** [comment]
image
https://devguide.python.org/versions/

File: _version_helper.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    hg.parse_archival,
+]
+
+
+def parse(root: str, config: Configuration) -> ScmVersion | None:
+    for maybe_parse in try_parse:
+        try:
+            parsed = maybe_parse(root, config)
+        except OSError as e:
+            log.warning("parse with %s failed with: %s", maybe_parse, e)
```
Comment (longjiquan) [inline _version_helper.py:46]: should be `maybe_parse.__name__`?

File: _version_helper.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        else:
+            if parsed is not None:
+                return parsed
+    else:
```
Comment (longjiquan) [inline _version_helper.py:50]: No `if` statement for this else branch.

**Review by longjiquan — CHANGES_REQUESTED** [review]

File: _version_helper.py (hunk 1 above)
Comment (XuanYang-cn) [reply _version_helper.py:46]: for `git.parse` maybe parse will give more info like
```
<function parse at 0x7fa03a40e5f0>
```

but __name__ would be just 
```
parse
```

**Pull request merged**